src/libsyntax/ast.rs

```diff
@@ -46,7 +45,7 @@ pub struct SyntaxContext(pub u32);
 /// An identifier contains a Name (index into the interner
 /// table) and a SyntaxContext to track renaming and
 /// macro expansion per Flatt et al., "Macros That Work Together"
-#[derive(Clone, Copy, Eq)]
+#[derive(Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
So, operator `==` and hash for `Ident` did unhygienic comparisons and now they do hygienic comparisons.
I'm not sure what unintended consequences it may have.
I remember derived implementations of `PartialEq` were used in couple of places for unhygienic comparisons of larger parts of AST including identifiers as their parts (maybe even expressions?), now the meaning of those comparisons changed.
Have you looked at where `==` for `Ident`s was previously used and how the meaning of that code changed?